src/sdk/analytics/index.tsx

```diff
@@ -1,13 +1,50 @@
+import type { AnalyticsEvent } from '@faststore/sdk'
 import { useAnalyticsEvent } from '@faststore/sdk'
 import type { PropsWithChildren } from 'react'
 
 if (typeof window !== 'undefined') {
   window.dataLayer = window.dataLayer ?? []
 }
 
+const GA_CLIENT_ID_KEY = 'main::store::ga::client_id'
+
+// https://gist.github.com/kindy/35ccd6c60b650e7fd261
+function generateClientId() {
+  return `${Math.floor(Math.random() * 0x7fffffff)}.${Math.floor(
+    Date.now() / 1000
+  )}`
+}
+
+const getClientId = () => {
+  let localClientId
+
+  try {
+    localClientId = localStorage.getItem(GA_CLIENT_ID_KEY)
+  } catch (e) {
+    console.error(e)
+  }
+
+  if (!localClientId) {
+    localClientId = generateClientId()
+    localStorage.setItem(GA_CLIENT_ID_KEY, localClientId)
+  }
+
+  return localClientId
+}
+
 export const AnalyticsHandler = ({ children }: PropsWithChildren<unknown>) => {
-  useAnalyticsEvent((event) => {
+  useAnalyticsEvent((event: AnalyticsEvent) => {
     window.dataLayer.push(event)
```

Comment:
why keeping it?

Reply:
Yeah, we can remove

Reply:
Do you have a tip on how to test the events layer, since those use the data layer?

Reply:
Also, I think we should have an easy way to see the event sequence.